core/camel-main/src/main/java/org/apache/camel/main/MainDurationEventNotifier.java

```diff
@@ -100,9 +94,12 @@ protected void doNotify(CamelEvent event) throws Exception {
             return;
         }
 
+        final boolean complete = event.getType() == CamelEvent.Type.ExchangeCompleted || event.getType() == CamelEvent.Type.ExchangeFailed;
```

Comment:
I wonder if you can optimize this to only compute complete and begin if maxMessages > 0 and maxIdelSeconds > 0

Reply:
Good idea! I think I can. I will double check tomorrow.